Ctrl+C is broken if pressed within Transactional (Quarkus 2.0.1.Final)

## Describe the bug
It seems that since Quarkus 2.0.1.Final, if Ctrl+C is pressed while in a `Transactional` block, quarkus:dev is not stopped. It is necessary to kill -9 the process.

### Expected behavior
quarkus:dev should stop when Ctrl+C is pressed, application should terminate.

### Actual behavior
When Ctrl+C is pressed while the application is in a `Transactional` block ^C is written to the terminal, but the application does not respond. Later (when the task completes or the transaction timeout passes) some exceptions are printed.


## To Reproduce
Use attached reproducer:
- Start it, with: `mvn quarkus:dev -Dquarkus.test.flat-class-path=true`
- Trigger the transaction: curl http://localhost:8080/fruits 
- Press Ctrl+C to stop quarkus:dev
- 

## Environment (please complete the following information):

### Output of `uname -a` or `ver`
`Linux a 5.12.9-1-MANJARO #1 SMP PREEMPT Thu Jun 3 14:56:42 UTC 2021 x86_64 GNU/Linux`

### Output of `java -version`
```
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)
```

### GraalVM version (if different from Java)

### Quarkus version or git rev

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
```
Apache Maven 3.8.1 (NON-CANONICAL_2021-04-26T21:52:54Z_root)
Maven home: /opt/maven
Java version: 11, vendor: Oracle Corporation, runtime: /opt/openjdk-11
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.12.9-1-manjaro", arch: "amd64", family: "unix"
```

## Additional context
Work if Quarkus version < 2.0

Application output including exceptions:
```
2021-07-07 17:08:41,021 INFO  [io.quarkus] (Quarkus Main Thread) hibernate-orm-quickstart 1.0-SNAPSHOT on JVM (powered by Quarkus 2.0.1.Final) started in 2.926s. Listening on: http://localhost:8080
2021-07-07 17:08:41,025 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-07-07 17:08:41,025 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [agroal, cdi, hibernate-orm, jdbc-postgresql, narayana-jta, resteasy, resteasy-jackson, smallrye-context-propagation, vertx]
Sleeping
Quick! Press Ctrl+C, now!
^C^C^C^CCan't get out, eh? :)
2021-07-07 17:09:01,332 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-0) HTTP Request to /fruits failed, error id: ac6faae5-e5c8-4b4f-814f-998b99777090-1: org.jboss.resteasy.spi.UnhandledException: java.lang.IllegalArgumentException
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:138)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:481)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException
	at io.quarkus.arc.impl.ArcContainerImpl.unwrap(ArcContainerImpl.java:757)
	at io.quarkus.arc.impl.EventImpl.createNotifier(EventImpl.java:139)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.quarkus.arc.impl.EventImpl.getNotifier(EventImpl.java:113)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:70)
	at io.quarkus.narayana.jta.runtime.CDIDelegatingTransactionManager.rollback(CDIDelegatingTransactionManager.java:117)
	at io.quarkus.narayana.jta.runtime.CDIDelegatingTransactionManager_Subclass.rollback$$superforward1(CDIDelegatingTransactionManager_Subclass.zig:415)
	at io.quarkus.narayana.jta.runtime.CDIDelegatingTransactionManager_Subclass$$function$$8.apply(CDIDelegatingTransactionManager_Subclass$$function$$8.zig:24)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:44)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(InvocationInterceptor_Bean.zig:521)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at io.quarkus.narayana.jta.runtime.CDIDelegatingTransactionManager_Subclass.rollback(CDIDelegatingTransactionManager_Subclass.zig:1240)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.endTransaction(TransactionalInterceptorBase.java:311)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:164)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:100)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:32)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:53)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:26)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:340)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at org.acme.hibernate.orm.FruitResource_Subclass.get(FruitResource_Subclass.zig:211)
	at org.acme.hibernate.orm.FruitResource_ClientProxy.get(FruitResource_ClientProxy.zig:155)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:408)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	... 15 more

2021-07-07 17:09:01,344 WARN  [io.qua.arc.impl] (executor-thread-0) An error occurred during delivery of the @BeforeDestroyed(RequestScoped.class) event: java.lang.NullPointerException
	at io.quarkus.arc.runtime.devconsole.EventsMonitor_Observer_notify_513ab14b479908baeab956aa73f8a6221325887f.notify(EventsMonitor_Observer_notify_513ab14b479908baeab956aa73f8a6221325887f.zig:98)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:300)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:288)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:230)
	at io.quarkus.arc.impl.RequestContext.fireIfNotEmpty(RequestContext.java:193)
	at io.quarkus.arc.impl.RequestContext.destroy(RequestContext.java:165)
	at io.quarkus.arc.impl.RequestContext.destroy(RequestContext.java:148)
	at io.quarkus.arc.ManagedContext.terminate(ManagedContext.java:34)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:153)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:481)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)

2021-07-07 17:09:01,345 WARN  [io.qua.arc.impl] (executor-thread-0) An error occurred during delivery of the @Destroyed(RequestScoped.class) event: java.lang.NullPointerException
	at io.quarkus.arc.runtime.devconsole.EventsMonitor_Observer_notify_513ab14b479908baeab956aa73f8a6221325887f.notify(EventsMonitor_Observer_notify_513ab14b479908baeab956aa73f8a6221325887f.zig:98)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:300)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:288)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:230)
	at io.quarkus.arc.impl.RequestContext.fireIfNotEmpty(RequestContext.java:193)
	at io.quarkus.arc.impl.RequestContext.destroy(RequestContext.java:173)
	at io.quarkus.arc.impl.RequestContext.destroy(RequestContext.java:148)
	at io.quarkus.arc.ManagedContext.terminate(ManagedContext.java:34)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:153)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:481)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)

```

--- a/extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/CDIDelegatingTransactionManager.java
+++ b/extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/CDIDelegatingTransactionManager.java
@@ -18,6 +18,8 @@ import javax.transaction.Transaction;
 import javax.transaction.TransactionManager;
 import javax.transaction.TransactionScoped;
 
+import org.jboss.logging.Logger;
+
 /**
  * A delegating transaction manager which receives an instance of Narayana transaction manager
  * and delegates all calls to it.
@@ -25,6 +27,9 @@ import javax.transaction.TransactionScoped;
  */
 @Singleton
 public class CDIDelegatingTransactionManager implements TransactionManager, Serializable {
+
+    private static final Logger log = Logger.getLogger(CDIDelegatingTransactionManager.class);
+
     private static final long serialVersionUID = 1598L;
 
     private final transient com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionManagerImple delegate;
@@ -113,13 +118,18 @@ public class CDIDelegatingTransactionManager implements TransactionManager, Seri
      */
     @Override
     public void rollback() throws IllegalStateException, SecurityException, SystemException {
-        if (this.transactionScopeBeforeDestroyed != null) {
-            this.transactionScopeBeforeDestroyed.fire(this.getTransaction());
+        try {
+            if (this.transactionScopeBeforeDestroyed != null) {
+                this.transactionScopeBeforeDestroyed.fire(this.getTransaction());
+            }
+        } catch (Throwable t) {
+            log.error("Failed to fire @BeforeDestroyed(TransactionScoped.class)", t);
         }
 
         try {
             delegate.rollback();
         } finally {
+            //we don't need a catch block here, if this one fails we just let the exception propagate
             if (this.transactionScopeDestroyed != null) {
                 this.transactionScopeDestroyed.fire(this.toString());
             }

